src/qibolab/instruments/qblox/cluster_qcm_bb.py

```diff
@@ -125,121 +109,119 @@ class ClusterQCM_BB(Instrument):
         lambda self, x: parent._set_device_parameter(parent.device.sequencers[sequencer], *parameter, value=x),
     )
 
-    def __init__(self, name: str, address: str, settings: dict):
+    def __init__(self, name: str, address: str, cluster: Cluster = None):
         """Initialises the instance.
 
         All class attributes are defined and initialised.
+
+        Args:
+            name(str): A unique name given to the instrument.
+            address: IP_address:module_number (the IP address of the cluster and module number)
+            cluster: the cluster to which the instrument belongs.
```

Comment:
Can we please add the correct type hint also in the docstring

Reply:
I would actually suggest avoiding type hints in the docstrings, just because they are now redundant with the actual hints, and it's more surface to be kept up to date (Sphinx is also fetching the parameter hints, so they will always be displayed).

In general, given the transition to explicit types, we do not necessarily need structured docstrings everywhere.
If possible, just keep few parameters in each function, and you can adopt a more conversational style (like it happens in Python docs, below some examples).
https://docs.python.org/3/library/shutil.html?highlight=shutils#shutil.copy2
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath
https://docs.python.org/3/library/subprocess.html#subprocess.run
(even though they do not have the explicit types :P)

Examples are always welcome :)

Reply:
In your opinion something like this can work?

```python

 def __init__(self, name: str, address: str, cluster: Cluster = None):
        """
        Initialize a Qblox QCM baseband module.

        Parameters:
        - name: An arbitrary name to identify the module.
        - address: The network address of the instrument, specified as "cluster_IP:module_slot_idx".
        - cluster: The Cluster object to which the QCM baseband module is connected.

        Example:
        --------
        To create a ClusterQCM_BB instance named 'qcm_bb' connected to slot 2 of a Cluster at address '192.168.0.100':

        >>> cluster_instance = Cluster("cluster","192.168.1.100", settings)
        >>> qcm_module = ClusterQCM_BB(name="qcm_bb", address="192.168.1.100:2", cluster=cluster_instance)
        """
```